pytorch_lightning/utilities/cli.py

```diff
@@ -336,6 +440,15 @@ def add_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
 
     def link_optimizers_and_lr_schedulers(self) -> None:
         """Creates argument links for optimizers and learning rate schedulers that specified a ``link_to``."""
+        if any(True for v in sys.argv for optim_name in OPTIMIZER_REGISTRY if f"--optimizer={optim_name}" in v):
```

Comment:
I would suggest modifying this so that it does not use two `for` loops which makes it over-complicated to understand. Could be:

```suggestion
        if any(re.sub('^--optimizer=', '', v) in OPTIMIZER_REGISTRY for v in sys.argv):
```

Reply:
This isn't strictly equivalent.